library/Rules/Video.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/*
+ * This file is part of Respect/Validation.
+ *
+ * (c) Alexandre Gomes Gaigalas <alexandre@gaigalas.net>
+ *
+ * For the full copyright and license information, please view the "LICENSE.md"
+ * file that was distributed with this source code.
+ */
+
+namespace Respect\Validation\Rules;
+
+class Video extends AbstractRule
+{
+    public function validate($input)
+    {
+        return !empty($input) && (
+            preg_match('/^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#\&\?]*).*/i', $input) ||
+            preg_match('/^.*(vimeo\.com\/)((channels\/[A-z]+\/)|(groups\/[A-z]+\/videos\/)|(video\/))?([0-9]+)/i', $input)
```

Comment:
They are validators URL:

Youtube - Accepts URL Youtuve
Vimeo - Accepts URL Vimeo
Video - Supports both URL

I would like to move to UrlYoutube?

Sorry my bad engilish.